pandas/tests/resample/test_resample_api.py

```diff
@@ -989,3 +989,27 @@ def test_resample_empty():
     )
     result = df.resample("8h").mean()
     tm.assert_frame_equal(result, expected)
+
+
+def test_asfreq_respects_origin_with_fixed_freq_all_seconds_equal():
+    # GH#62725: Ensure Resampler.asfreq respects origin="start_day"
+    # when all datetimes share identical seconds values.
+    idx = [
+        datetime(2025, 10, 17, 17, 15, 10),
+        datetime(2025, 10, 17, 17, 16, 10),
+        datetime(2025, 10, 17, 17, 17, 10),
+    ]
+    df = DataFrame({"value": [0, 1, 2]}, index=idx)
+
+    result = df.resample("1min", origin="start_day").asfreq()
+
+    # Expected index: match dtype while preserving freq
+    exp_idx = pd.DatetimeIndex(
+        date_range("2025-10-17 17:15:00", periods=3, freq="min").astype(
+            result.index.dtype
+        ),
```

Comment:
Could you pass a list of `Timestamp`s to the `pd.DatetimeIndex` constructor and pass the `dtype` as well? (For context, tests should minimize the use of extraneous APIs where possible.)